src/main/java/org/truffleruby/language/methods/Arity.java

```diff
@@ -90,6 +94,42 @@ public int getArityNumber() {
         return keywordArguments;
     }
 
+    public ArgumentDescriptor[] toAnonymousArgumentDescriptors() {
+        ArrayList<ArgumentDescriptor> descs = new ArrayList<>();
```

Comment:
Style suggestion: `List<ArgumentDescriptor> descriptors`, alternatively you might be able to avoid the List altogether (not sure if that's worth the effort.

Reply:
Yeah even though it's relatively easy to compute the length here, I think it's nicer to not have to keep track of the position manually with an extra counter variable.